app/controllers/permissions_controller.rb

```diff
@@ -339,21 +342,23 @@ def construct_request_object(provider)
       collection_identifier['entry_titles'] = entry_titles
     end
 
-    @collection_access_value = params[:collection_access_value] || {}
-    @collection_access_value.each do |key, val|
-      if val.blank?
-        @collection_access_value.delete(key)
-      elsif val == 'true'
-        @collection_access_value[key] = true
-      else
-        @collection_access_value[key] = val.to_f
+    if collection_applicable
+      @collection_access_value = params[:collection_access_value] || {}
+      @collection_access_value.each do |key, val|
+        if val.blank?
+          @collection_access_value.delete(key)
+        elsif val == 'true'
```

Comment:
I am running into this issue: http://stackoverflow.com/questions/1711547/how-to-check-if-a-param-is-true-or-false. Is there a better way to handle it?